### PR TITLE
factor out performance profile coordinates from performance_profile

### PR DESCRIPTION
**Pull request by lkapelevich**
addresses #49 

@abelsiqueira this does what I was after, but I didn't export or document `performance_profile_data`, let me know if you would like me to. Also this allocates vectors of vectors to plot, I assumed it's not a big deal.

### CODE REVIEW

File: src/performance_profiles.jl [hunk 1]
```diff
@@ -54,38 +90,17 @@ function performance_profile(T :: Array{Float64,2}, labels :: Vector{AbstractStr
...
-      k += 1
-      xidx[k] = findlast(rs .<= rv)
-      rv = max(rs[xidx[k]] + sampletol, rs[xidx[k]+1])
-    end
-    xidx[k+1] = length(rs)
-    xidx = xidx[xidx .> 0]
-    xidx = unique(xidx) # Needed?
-    if length(linestyles) > 0
-      kwargs[:linestyle] = linestyles[s]
-    end
```
Comment (abelsiqueira) [inline src/performance_profiles.jl:83]: This went missing

File: test/runtests.jl [hunk 2]
```diff
@@ -7,4 +7,3 @@ T = 10 * rand(25, 3)
 performance_profile(T, ["a", "b", "c"], title="Test Profile", linestyles=[:solid, :dash, :dot])
 H = rand(25, 4, 3)
 data_profile(H, ones(10), ["a", "b", "c"], title="Test Profile")
-
```
Comment (abelsiqueira) [inline test/runtests.jl:10]: No need to change this

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks for the PR. I would like some docs, please. No need to `export` it though. I made a small comment.

@dpo, do you want to review this?

**Comment by dpo** [comment]
Thanks @lkapelevich !

**Review by abelsiqueira — APPROVED** [review]

**Review by dpo — APPROVED** [review]

**Comment by abelsiqueira** [comment]
There is an issue with the docs, so I'm testing locally

**Comment by abelsiqueira** [comment]
Thanks!

**Comment by lkapelevich** [comment]
Thank you! Would you mind tagging?

**Comment by abelsiqueira** [comment]
Will do

**Pull request merged**